torch/_inductor/ir.py

```diff
@@ -3882,16 +3884,18 @@ def add_submodule(self, block, prefix):
         self.submodules[name] = block
         return name
 
-    def add_indirect(self):
+    def add_indirect(self, size):
         name = f"indirect{len(self.indirect_vars)}"
         var = sympy_symbol(name)
         self.indirect_vars.append(var)
+        self.indirect_max_sizes.append(size)
         return var
 
     def replace_indirect(self, old, new):
         """Swap in a variable used in indirect indexing"""
         if str(old) == str(new):
             return
+        self.indirect_new[old] = new
```

Comment:
this is weird
1) if we never really replace_indirect (old is always same as new), `indirect_new` will be empty
2) in current impl, it looks like `old` should disappear completely after replacement, yet now we are using it to index into `indirect_new`, what's going on?

Reply:
So, I added this because there are times where `replace_indirect` is called a few times with the same `old` but different `new`s. As such, if I simply had a dictionary `var -> sizes`, then here I'd need to update the var by a newvar, but then when `replace_indirect` is called with `var, newvar2`, I wouldn't have the `var` information any more. The current way, I'm simply keeping the last replacement and that's it.